Source/MQTTnet/Server/Internal/MqttRetainedMessagesManager.cs

```diff
@@ -135,5 +137,14 @@ public async Task ClearMessages()
                 await _eventContainer.RetainedMessagesClearedEvent.InvokeAsync(EventArgs.Empty).ConfigureAwait(false);
             }
         }
+
+        private static bool SequenceEqual(ArraySegment<byte> source, ArraySegment<byte> target)
+        {
+#if NETCOREAPP3_1_OR_GREATER || NETSTANDARD2_1
+            return source.Count == target.Count && source.AsSpan().SequenceEqual(target);
```

Comment:
```suggestion
            return source.AsSpan().SequenceEqual(target);
```
`SequenceEqual` also does a length check before doing the actual compare, so no need to have that check twice.